go/pkg/cas/client.go

```diff
@@ -68,31 +67,42 @@ type ClientConfig struct {
 	// FileIOSize is the size of file reads.
 	FileIOSize int64
 
-	// FindMissingBlobsBatchSize is the maximum number of digests to check in a
-	// single FindMissingBlobs RPC.
-	FindMissingBlobsBatchSize int
+	// FindMissingBlobs is configuration for FindMissingBlobs RPCs.
+	// FindMissingBlobs.MaxSizeBytes is ignored.
+	FindMissingBlobs RPCConfig
 
-	// BatchUpdateBlobsConcurrency is the maximum number of BatchUpdateBlobs RPCs
-	// at the same time.
-	BatchUpdateBlobsConcurrency int
-
-	// MaxBatchTotalSizeBytes is the maximum number of blobs to upload/download
-	// in a single BatchUpdateBlobs/BatchReadBlobs RPC.
-	//
-	// If server capabilities are consulted (see IgnoreCapabilities), then
-	// this value is capped to the server-imposed limit.
-	// Thus Client.ClientConfig.MaxBatchTotalSizeBytes might be less than what
-	// was specified in NewClientWithConfig.
-	MaxBatchTotalSizeBytes int
+	// BatchUpdateBlobs is configuration for BatchUpdateBlobs RPCs.
+	BatchUpdateBlobs RPCConfig
 
 	// RetryPolicy specifies how to retry requests on transient errors.
 	RetryPolicy retry.BackoffPolicy
 
 	// IgnoreCapabilities specifies whether to ignore server-provided capabilities.
 	// Capabilities are consulted by default.
 	IgnoreCapabilities bool
+}
+
+// RPCConfig is configuration for a particular CAS RPC.
+// Some of the fields might not apply to certain RPCs.
+type RPCConfig struct {
+	// Concurrency is the maximum number of RPCs at a time.
+	Concurrency int
+
+	// MaxSizeBytes is the maximum size of the request/response, in bytes.
+	MaxSizeBytes int
+
+	// MaxItems is the maximum number of blobs/digests per RPC.
+	// Applies to batch RPCs.
+	MaxItems int
 
-	// TODO(nodir): add per-RPC timeouts.
+	// Timeout is the maximum duration of the RPC.
+	Timeout time.Duration
+
+	sem *semaphore.Weighted
```

Comment:
It's slightly awkward that there's a semaphore in a config struct? Would moving it to the client make sense?

Reply:
This was intentional, but undone now.

Reply:
Sorry not following, what is undone? My point is that it seems like an antipattern that the config struct would need to be initialized and would have state.

Reply:
sorry, just noticed that `git push` was still waiting for me to enter my ssh key password.. Still getting used to GitHub.
PTAL again.

I said "undone" because I moved semaphores to RPCConfig initially, and the undid that after your request.

Reply:
Ah now it makes sense! Thanks.